Doc/library/exceptions.rst

```diff
@@ -965,6 +965,10 @@ their subgroups based on the types of the contained exceptions.
          def derive(self, excs):
             return Errors(excs, self.exit_code)
 
+   Like :exc:`ExceptionGroup`, any subclass of :exc:`BaseExceptionGroup` which
+   is also a subclass of :exc:`Exception` can only wrap instances of
+   :exc:`Exception`.
```

Comment:
As written, this suggests that `class MyEG(BaseExceptionGroup, ValueError):` should be unable to wrap anything but `ValueError`s!

That may in fact be the intention, and it wouldn't break the `trio.MultiError`-as-`BaseExceptionGroup` deprecation pathway, but it's not the implementation and the tests don't distinguish these cases.

Reply:
> As written, this suggests that class MyEG(BaseExceptionGroup, ValueError): should be unable to wrap anything but ValueErrors!

Does it? In the doc the Exception becomes a hyperlink.

Reply:
But feel free to make a PR with better wording.

Reply:
I expect some users to read that as "if a subclass of BaseExceptionGroup is also a subclass of X, it can only wrap instances of X"; I'll open a PR to rephrase as "if it's also a subclass of Exception, it can't wrap BaseExceptions".

But then we're still treating `Exception` as particularly special for subclasses, which feels a bit strange.

Reply:
Zac, I'm confused why you think the sentence as written could be interpreted that way. It says `Exception` twice. In this context that is very clearly a reference to the specific built-in class named `Exception` (it even links there), not a "free variable" that must refer to the same class in both places.

Regardless, even if we clear that up, it feels like it would be expensive to inspect the class hierarchy to find other subclasses of `BaseException` and try to constrain the arguments to inherit from those. And that's definitely not what the PEP intended. We want the common case (where you're using one of the two built-in EG classes) to be foolproof, and other cases _possible_.

Reply:
Apologies for the confusion, that's on me 😞

I do think the behavior in this PR is good, and I'll open a PR with some additional tests and maybe docs to distinguish it from the case that I was wondering about.  Thanks again to both of you for everything.